Source/PLCrashReporter.m

```diff
@@ -746,7 +746,9 @@ - (NSData *) generateLiveReportWithThread: (thread_t) thread error: (NSError **)
         goto cleanup;
     }
 
-    data = [NSData dataWithContentsOfFile: [NSString stringWithUTF8String: path]];
+    data = [NSData dataWithContentsOfFile:[NSString stringWithUTF8String: path]
+                                  options:NSDataReadingMappedAlways | NSDataReadingUncached
```

Comment:
As far as I understand, mapped files should only be used if the file is guaranteed to exist for the duration of the data object’s existence. But there is a check below that assumes that the file can suddenly deleted.

Also, reading file with mapping option won't actually read the data until it actually needed. In this case it is placed here to check reading the crash report file. So, I don't think that this change makes sense here.

Reply:
> Also, could you please share use case when you need "many live reports in a short period of time"?
In our app we have a non-fatal reporting API that is a wrapper on top of `generateLiveReportWithThread`. Some developers put it into the piece of code that in some cases was called too often (several times per second) and thus creating a memory pressure with a lot of short living heavy objects that caused an out of memory exception. While it was definitely a bad usage of API, it took us many dev hours in order to identify and fix this issue in our code because OOM exception were not related to PLCrashReporter at all. Instead the app was crashing whenever it allocates a big chunk of memory at once. So by adding this change we'll avoid similar issues in the future

Reply:
> As far as I understand, mapped files should only be used if the file is guaranteed to exist for the duration of the data object’s existence. But there is a check below that assumes that the file can suddenly deleted.

This won't be an issue at all as an open file is remain accessible to all the readers even after it is deleted. It becomes physically deleted only when all the readers close the file handle